Content.Server/GameTicking/GameTicker.Spawning.cs

```diff
@@ -241,10 +262,43 @@ private void SpawnPlayer(IPlayerSession player, HumanoidCharacterProfile charact
                 _chatManager.DispatchServerMessage(player, Loc.GetString("latejoin-arrivals-direction"));
             }
 
+            return mob;
+        }
+
+
+        // Called by a system which is currently spawning ghost-role players deliberately, like a tourist event
+        public bool SpawnPlayerAgain(IPlayerSession player, EntityUid station, string jobId)
+        {
+            // Can't spawn players with a dummy ticker!
+            if (DummyTicker)
+                return true;
+
+            var data = player.ContentData();
+            DebugTools.AssertNotNull(data);
+            if (data == null)
+                return false;
+
+            ResolveStation(ref station);
+
+            // Figure out job restrictions
+            var jobBans = _banManager.GetJobBans(player.UserId);
+            if (jobBans == null || jobBans.Contains(jobId))
+                return false;
+
+            if (!_playTimeTrackings.IsAllowed(player, jobId))
+                return false;
```

Comment:
since you're splitting out the resolvestation/spawnandattachmind stuff, you may as well split this part out into its own function so its not duplicated